### PR TITLE
Update setup-go and golangci-lint actions

### PR DESCRIPTION
**Pull request by dsimansk**
Per title.

/cc @matzew

### CODE REVIEW

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
                 -level="error"
 
       - name: Go Lint - knative-operator
-        uses: golangci/golangci-lint-action@v2
+        uses: golangci/golangci-lint-action@v3
```
Comment (dsimansk) [inline .github/workflows/validate.yaml:115]: >v3.0.0+ requires explicit setup-go installation step prior to using this action: uses: actions/setup-go@v2.
The skip-go-installation option has been removed.

Fixes the issue with too fresh go version being installed for golangci-lint runs.

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -112,7 +112,7 @@ jobs:
...
         with:
           version: v1.43
```
Comment (dsimansk) [inline .github/workflows/validate.yaml:117]: @matzew @skonto @pierDipi should we stick with whatever upstream actions use or just go latest by default?

Comment (skonto) [reply .github/workflows/validate.yaml:117]: I updated to latest #1465 ;)

Comment (pierDipi) [reply .github/workflows/validate.yaml:117]: I have no real opinion as long as it works and is useful.

**Comment by dsimansk** [comment]
nevermind :)
/close

**Pull request closed without merging**